CMakeLists.txt

```diff
@@ -68,11 +73,19 @@ endif()
 # Find OpenSSL
 ###############################################################################
 
+option(OpenSSL_REQUIRED "Enable or disable the requirement of OpenSSL" ON)
 find_package(OpenSSL)
-include_directories(${OPENSSL_INCLUDE_DIR})
-set(LIBS ${LIBS} ${OPENSSL_LIBRARIES})
-if(NOT OPENSSL_FOUND)
-    add_definitions("-DCONF_NO_TLS")
+if(OPENSSL_FOUND)
+    include_directories(${OPENSSL_INCLUDE_DIR})
+    set(LIBS ${LIBS} ${OPENSSL_LIBRARIES})
+else()
+    if(OpenSSL_REQUIRED)
+        message(FATAL_ERROR "OpenSSL NOT found: use `-DOpenSSL_REQUIRED=OFF` to build without SSL support")
+    else()
+        if(NOT OPENSSL_FOUND)
```

Comment:
It is equal to the feature in xmr-stak-nvidia and xmr-stak-cpu.

It is possible to disable `OpenSSL` with `-DCMAKE_DISABLE_FIND_PACKAGE_OpenSSL=TRUE`. This is the [default way for CMake](https://cmake.org/cmake/help/v3.2/manual/cmake-packages.7.html#id9).  This is only possible if `find_package` is used. It is not possible for `microhttpd`.